#483 add CheckboxCell renderer for boolean columns

junaidzm13 opened the pull request:
- also adds this renderer as the default for boolean columns in case no renderer is specified.

junaidzm13 committed:
#483 add CheckboxCell renderer for boolean columns

- also adds this renderer as the default for boolean columns in
  case no renderer is specified.

heswell commented on vuu-ui/packages/vuu-data-test/src/simul/simul-schemas.ts:
```diff
@@ -22,6 +22,9 @@ export const schemas: Readonly<Record<SimulTableName, Readonly<TableSchema>>> =
         { name: "isin", serverDataType: "string" },
         { name: "lotSize", serverDataType: "int" },
         { name: "ric", serverDataType: "string" },
+        { name: "price", serverDataType: "double" },
```
price was excluded by design here. It's really there to provide a seed value for price generation, for the prices that appear inn the prices table.

heswell commented on vuu-ui/packages/vuu-table/src/table-next/cell-renderers/checkbox-cell/CheckboxCell.tsx:
```diff
@@ -0,0 +1,39 @@
+import { dispatchCustomEvent, registerComponent } from "@finos/vuu-utils";
+
+export const CheckboxCell: React.FC<TableCellRendererProps> = memo(
+  ({ column, columnMap, onCommit = WarnCommit, row }) => {
```
by using an anonymous 'fat arrow' function like this, the component will appear as 'Anonymous' in the React Devtools. Better to use a named function inline
```
memo(function CheckboxCell(){ ...
``` 

OR explicitly assign a static displayName property
```
CheckboxCell.displayName = "CheckboxCell"
```

heswell commented on vuu-ui/packages/vuu-table/src/table-next/cell-renderers/checkbox-cell/CheckboxCell.tsx:
```diff
@@ -0,0 +1,39 @@
+    );
+
+    return (
+      <React.Fragment>
```
no need for the fragment here

heswell commented on vuu-ui/packages/vuu-ui-controls/src/list/CheckboxIcon.css:
We should move this icon out of List if its being sharewd across components, but leave it here and commit for now. I'll have a think about where it might live

heswell reviewed (changes requested):
Looks great, a few very minor tweaks to make, than we'll merge

junaidzm13 committed:
#483 small code clean-up and bug fix in vuu-data-test

- also reverts the addition of prices column in simul instruments
  table in the previous commit.
